src/Database/Database.php

```diff
@@ -0,0 +1,1638 @@
+<?php
+
+namespace Friendica\Database;
+
+use Friendica\Core\Config\Cache\IConfigCache;
+use Friendica\Core\System;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Profiler;
+use mysqli;
+use mysqli_result;
+use mysqli_stmt;
+use PDO;
+use PDOException;
+use PDOStatement;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @class MySQL database class
+ *
+ * This class is for the low level database stuff that does driver specific things.
+ */
+class Database
+{
+	private $connected = false;
+
+	/**
+	 * @var IConfigCache
+	 */
+	private $configCache;
+	/**
+	 * @var Profiler
+	 */
+	private $profiler;
+	/**
+	 * @var LoggerInterface
+	 */
+	private $logger;
+	private $server_info    = '';
+	private $connection;
+	private $driver;
+	private $error          = false;
+	private $errorno        = 0;
+	private $affected_rows  = 0;
+	private $in_transaction = false;
+	private $in_retrial     = false;
+	private $relation       = [];
+	private $db_serveraddr  = '';
+	private $db_user        = '';
+	private $db_pass        = '';
```

Comment:
The database password shouldn't be stored in plaintext in the object, in case there's a verbose Exception output that would publicly spill the bean. We should use https://packagist.org/packages/paragonie/hidden-string instead.